src/Compilers/CSharp/CSharpAnalyzerDriver/CSharpDeclarationComputer.cs

```diff
@@ -14,14 +14,14 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal class CSharpDeclarationComputer : DeclarationComputer
     {
-        public static void ComputeDeclarationsInSpan(SemanticModel model, TextSpan span, bool getSymbol, List<DeclarationInfo> builder, CancellationToken cancellationToken)
+        public static void ComputeDeclarationsInSpan(SemanticModel model, TextSpan span, bool getSymbol, ImmutableArray<DeclarationInfo>.Builder builder, CancellationToken cancellationToken)
```

Comment:
please wrap long lines.  On github this was not at all clear that this was a Builder and not an ImmutableArray.

Reply:
use split view. it will show long line.

Reply:
i would prefer our code be usable on github in either view :)

Reply:
well. not sure when there is a way to see it on GitHub without too much trouble. that feels like something github should fix not code to change to fit into github.

Reply:
> without too much trouble.

it is a lot of trouble.  because now you have to use split view instead of the unified view.

it also affects anyone who wants to ever look at teh code on github *outside* of diff-view as it will scroll off the right.

> that feels like something github should fix not code to change to fit into github.

I disagree.  We're an open source project that uses github extensively as the medium for everyone to collaborate.  We should be cognizant of that and should try to be good citizens here.  If the tooling ever changes, we can recalibrate.  But we should not expect the rest of the ecosystem to bend to us.  We should be considerate and be good to the rest of hte ecosystem. That's just good project stewardship.

Reply:
well, if everyone agree on that, we should just create a tool to format everything to that certain set and make sure we never violate that. and fix them all at once. otherwise, just picking here and there doesnt make sense to me.

that tool should be quite easy to create as analyzer/fixer. if you can pursue all roslyn team to adapt that analyzer/fixer, I dont have problem on that, otherwise. I dont think it is that much issue.

Reply:
> otherwise, just picking here and there doesnt make sense to me.

I'm not "picking here and there".  I'm reading a PR and i'm pointing out the parts of the PR that made less sense because information was hidden by default.  When that happens, i point it out so it can be addressed.  I'm also totally for automated approaches to that.  But that has no bearing on if we should still make our PRs better in the meantime.  Even if we don't have an automated approach, when we make code changes we should do our best to make those changes good for all these important use cases (PRs, reading on gitub, etc. etc.)

Honestly, we've spent more time arguing about this than would take to simply hit "enter"-"commit"-"push".  I find it amazing that instead of just makign things better we'd rather spend more energy to just stay in the same poorer state that we started in.

Reply:
> that feels like something github should fix not code to change to fit into github.

It's not just a github issue. My laptop has limited screen size as well and that's true whether i'm looking at the code on github or in VS.

Reply:
hmmm.. so for people who has even less screen size, they can argue line  should be wrapped in even smaller length?

like I said, split view or github providing better scrolling or inline diff providing wrapped line or github not restricting width of code, seems all better solution than wrapping the line in code.

for me, the argument is strange since argument is based on something that is not fixed. such as current UI of a tool or screen size. and requires change in code.

so, if someone buy a new laptop or UI change, one should go and re-arrange all code?

Reply:
> hmmm.. so for people who has even less screen size, they can argue line should be wrapped in even smaller length?

Yes.  And we can make a reasonable determination for what's an acceptable width that satisfies the needs of the team without being too cramped.  IMO, 120 seems to be fine.  It works for github and our tools.  Seems to fit on nearly everyone's screen, and gives enough flexibility to the dev to write code without being so cramped that they're just continually wrapping.

Reply:
> so, if someone buy a new laptop or UI change, one should go and re-arrange all code?

There can be suitable minimums we can accept that work for nearly everyone.  For example, ew can expect that no one is going to be working at 640x480.  And, even if there was, it's ok for that extreme outlier to not be catered to.  There's no need to make reductive arguments when we can just make common sense decisions about what coding patterns sensibly work for the majority of the people working on Roslyn and using github as the primary collaboration medium.

Reply:
I think it is bad for both size of parties. it will make harder for github to change UI since code is hardcoded to fit into current UI of github and code is set to prefer certain UI of a tool.

for people who use other tool for diff say codeflow, with 4k 30 inch monitor, It will be like why the heck all code is wrapped into this small line when I have huge horizontal space left,.

Reply:
it's actually 130, that should be more than enough

Reply:
for me, this feels like the old time where people used text editor and things to move to word processor.

in old text editor time, you put "enter" to the text itself to go new line and to make it look good in the text editor. in another word, text included formatting. and word processor came along and tool start to do formatting and text only has care about content.

there was a lot of arguments at the time, and now years later, word processor won the fight since presentation is not something fixed and different form factor/tool/hardware require different formatting but content remain same.

this same thing happened with html. initial html had content and formatting together, now they are separated.

I believe it is same for mark down people love nowadays.

I dont see anything different for code. if presentation or formatting is issue, than tool is the problem.

Reply:
> it's actually 130, that should be more than enough

this is not true. I prefer big monitor. I just can't code in small monitor. so I don't like unnecessary wrapped lines since I have huge horizontal spaces. I do simple code review from github but anything big in other tool (code flow), so horizontal space is not an issue.

so wrapped line actually go against my workflow. is my workflow more important than others? no idea.

Reply:
> for people who use other tool for diff say codeflow, with 4k 30 inch monitor, It will be like why the heck all code is wrapped into this small line when I have huge horizontal space left,.

This is how most non-individual projects work.   And the reason here is the same reason as all those other ones: because it's a collaborative endeavor.  So we code accordingly.  

For your own projects you can code however you want.  For projects that have lots of developers, and ones that funnel through common tools like github, we should be considerate and code accordingly.

>  but anything big in other tool (code flow), so horizontal space is not an issue.

But it's a problem in github.  And github is the choice we've made as a team as to how we collaborate on this project.

> is my workflow more important than others? no idea.

No.  it is not.  That's the idea of a team project.  In your own forks and your own projects you can do whatever you want.  But for PRs to Roslyn itself, we should be acting in a manner that is considerate of the community and the tooling we've decided on for how we interact with that community.  For better or for worse, that is github and that's not going to change any time soon.  If/when we ever decide on something else, we can consider alternate approaches here.

Reply:
>  if presentation or formatting is issue, than tool is the problem.

It is our responsibility to create code that works well with the tools we chosen and the ecosystem and community we're a part of.  We can always try to make the tools better.  And if they get there, we can adjust accordingly.  However, in the interim period *until* that happens, we should not punt on this just because the tools are limited.  We should acknowledge and gracefully respect that that's where things are now instead of coding in a way that works poorly with these tools, which then negatively impacts the community.

Reply:
> for me, this feels like the old time where people used text editor and things start moving to word processor.

Sure. And if we get to a point where the equivalent tools exist that operate like the "word processor" in your analogy *and* enough of our community has moved to them *then* we can change how we code accordingly.  However, *until then*, these are the tools we have, and we should not make the experience poor in them.  It's just inconsiderate.

Reply:
I guess that's where I dont agree. 

I believe people who use bigger monitor is larger than number of people who use smaller monitor. (I believe, in very large number more) and when you use bigger monitor, spit view works completely fine. and non wrapped line works completely fine. in large monitor, split view uses full width of screen. so it not even to fit to github UI, it is to fit to some number of people who uses small screen + use inlined diff view. and don't want to ever use split view.

except those number of people, long line works fine.

Reply:
> except those number of people

Those people are using the standard and *default* view of github.  So yes, you are correct, it's fine unless you happen to be using the default.  :)

Reply:
> I believe people who use bigger monitor is larger than number of people who use smaller monitor. 

It's not the size of the monitor.  It's the resolution.  And even recent telemetry data (like 6-9 months old) showed the vast majority of devs were 1080p or less**.  I'm currently running 1080p, and i cannot see these lines.  

--

Other hardware surveys (like steam's) also show the vast majority of users are around this as well.  But MS has this info already and knows that above 1080p is actually the large minority here.

Reply:
@dotnet/roslyn-ide @dotnet/roslyn-analysis any opinion? if all think same, I can create simple analyzer that will issue warning if any line is longer than 130 characters. so that people can find that one as they use code.

fixer probably not issue since how to wrap is different per case by case. but we can identifies all those cases and make the change once for all. and don't think about it again.

Reply:
@CyrusNajmabadi  point taken , @heejaechang  can you pls file a separate issue for this and we can priorities it.

Reply:
> if all think same, I can create simple analyzer that will issue warning if any line is longer than 130 characters

I would have a passionate objection to this. I even blocked attempts to add it as a configurable rule to StyleCop Analyzers.

Reply:
@sharwell not even as a suggestion? warning is too aggressive but having a suggestion at the character after the column limit doesn't seem like a bad idea. what is your objection to that?